### ISSUE TITLE
file does not process: detect doc structure, captions, bibRef ActaEntomolSlov .16.2.105-116.pdf

### ISSUE THREAD

**Issue by myrmoteras** [gsautter/goldengate-imagine#844]
This is an oddball of a file.
* After decoding, the detect document structure does not work, unless the first section of the document is change from header to main text
* bibRef are not detected, and if I mark them up manually, they are not processed
* the captions are set as text stream type "caption", but the caption does not seem to be tagged (not green in many cases)
![image](https://user-images.githubusercontent.com/4609956/72134514-97944e00-3384-11ea-8768-ef3fa621c21e.png)
 
the IMF is here  https://drive.google.com/open?id=1rdZa4MWyeqH2Ggl60hXfsZ7sR8llHjCA

**Comment by gsautter** [comment]
The one thing that jumps my eye about this IMF is that all the numbers are shattered, one individual word per digit. This explains pretty much all of the failures you report:
- the references fail to detect because the presence of a four-digit year serves as a filter and years are in individual digits, which also throws off RefParse
- captions fail to detect because the figure numbers are split just like the years ... the patterns expect one number but end up finding two (the individual digits of the actual number)
- the same problem occurs for page numbers

The captions failing to annotate on "Mark Caption" is something I cannot reproduce ... they annotate just fine.
After joining all the years, the bibliography also detects fine, only RefParse has its fair share of problems with the title and journal name being separated by a colon, which is a pretty rare punctuation style and not even consistently used in this article, and also with several of the publication dates given between volume number and pagination.
Heading detection might well have a problem with the headings being merely bold, but no larger than the document main text (font size 11 throughout)
Apart from that, the mix of languages (English, Slovenian?, and French) in such a short text may well mess with frequency analyses, etc.

Now the exciting part: finding out just _why_ all those numbers got separated in the first place ...

**Comment by gsautter** [comment]
All of the technical stuff aside: Why, exactly, are we interested in this article?

**Comment by gsautter** [comment]
Figured out the numbers ... this PDF renders one character per command (extremely inefficient, but works), which initially ends all the individual characters in separate words. After topological sorting, horizontally adjacent _letters_ are merged if close enough, but turns out this was intentionally blocked for _digits_ ... after deactivating that restriction, the numbers come up fine. Just not sure why the PDF decoding facilities had this restriction in the first place, tending to think I introduced it at some point in the early days to prevent havoc in the presence of inaccuracies that have been fixed since.
